components/ActionSheet/src/MDCActionSheetController.m

```diff
@@ -0,0 +1,357 @@
+  _header.mdc_adjustsFontForContentSizeCategory = self.mdc_adjustsFontForContentSizeCategory;
+
+  [self.view addSubview:_tableViewController.tableView];
+  [self.view addSubview:_header];
+}
+
+- (void)viewWillLayoutSubviews {
+  [super viewWillLayoutSubviews];
+
+  if (needsPreferredContentSizeUpdate == false) {
+    [self layoutViews];
+  }
+}
+
+- (void)layoutViews {
+  CGFloat headerHeight = [_header sizeThatFits:self.view.bounds.size].height;
+  CGFloat width = CGRectGetWidth(self.view.bounds);
+  CGRect tableFrame = _tableViewController.tableView.frame;
+  tableFrame.size.width = width;
+  _tableViewController.tableView.frame = tableFrame;
+  [_tableViewController.tableView reloadData];
+  [_tableViewController.tableView setNeedsLayout];
+
+  /// We need this call to `layoutIfNeeded` to get the correct contentSize for the table
+  [_tableViewController.tableView layoutIfNeeded];
+  CGFloat tableHeight = _tableViewController.tableView.contentSize.height;
+  CGFloat height = headerHeight + tableHeight;
+  tableFrame = _tableViewController.tableView.frame;
+  tableFrame.origin.y = headerHeight;
+  _tableViewController.tableView.frame = tableFrame;
+  self.preferredContentSize = CGSizeMake(width, height);
+  needsPreferredContentSizeUpdate = true;
+}
+
+- (void)viewWillAppear:(BOOL)animated {
+  [super viewWillAppear:animated];
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+  self.mdc_bottomSheetPresentationController.delegate = self;
+#pragma clang diagnostic pop
+
+  self.mdc_bottomSheetPresentationController.dismissOnBackgroundTap =
+      _transitionController.dismissOnBackgroundTap;
+  [self.view layoutIfNeeded];
+}
+
+- (BOOL)accessibilityPerformEscape {
+  if (!self.dismissOnBackgroundTap) {
+    return NO;
+  }
+  [self dismissViewControllerAnimated:YES completion:nil];
+  return YES;
+}
+
+- (void)preferredContentSizeDidChangeForChildContentContainer:(id<UIContentContainer>)container {
+  [super preferredContentSizeDidChangeForChildContentContainer:container];
+
+  [self.presentationController
+      preferredContentSizeDidChangeForChildContentContainer:self];
+}
+
+- (UIScrollView *)trackingScrollView {
+  return _transitionController.trackingScrollView;
+}
+
+- (void)setTrackingScrollView:(UIScrollView *)trackingScrollView {
+  _transitionController.trackingScrollView = trackingScrollView;
+}
+
+- (BOOL)dismissOnBackgroundTap {
+  return _transitionController.dismissOnBackgroundTap;
+}
+
+- (void)setDismissOnBackgroundTap:(BOOL)dismissOnBackgroundTap {
+  _transitionController.dismissOnBackgroundTap = dismissOnBackgroundTap;
+  self.mdc_bottomSheetPresentationController.dismissOnBackgroundTap = dismissOnBackgroundTap;
+}
+
+- (void)viewWillTransitionToSize:(CGSize)size withTransitionCoordinator:
+    (id<UIViewControllerTransitionCoordinator>)coordinator {
+  [super viewWillTransitionToSize:size withTransitionCoordinator:coordinator];
+  
+  CGRect frame = self.view.frame;
```

Comment:
I believe you may want to use the coordinator to animate the change of these properties alongside the orientation change.